dbms/programs/server/MySQLHandler.cpp

```diff
@@ -266,29 +267,56 @@ void MySQLHandler::comPing()
 }
 
-void MySQLHandler::comQuery(ReadBuffer & payload)
-{
-    bool with_output = false;
-    std::function<void(const String &)> set_content_type = [&with_output](const String &) -> void {
-        with_output = true;
-    };
+void MySQLHandler::comQuery(ReadBuffer &payload) {
+    std::string query = std::string(payload.position(), payload.buffer().end());
+
+    // This is a workaround in order to support adding ClickHouse to MySQL using federated server.
+    // As Clickhouse doesn't support these statements, we just send OK packet in response.
+    if (
+        (0 == strncasecmp("SET NAMES", query.c_str(), 9))
+        ||
+        (0 == strncasecmp("SET character_set_results", query.c_str(), 25))
+        ||
+        (0 == strncasecmp("SET FOREIGN_KEY_CHECKS", query.c_str(), 22))
+        ||
+        (0 == strncasecmp("SET AUTOCOMMIT", query.c_str(), 14))
+        ||
+        (0 == strncasecmp("SET SESSION TRANSACTION ISOLATION LEVEL", query.c_str(), 39))
+    ) 
+    {
+        packet_sender->sendPacket(OK_Packet(0x00, client_capability_flags, 0, 0, 0), true);
+    } 
+    else 
+    {
+        bool with_output = false;
+        std::function<void(const String &)> set_content_type = [&with_output](const String &) -> void {
+            with_output = true;
+        };
 
-    const String query("select ''");
-    ReadBufferFromString empty_select(query);
+        String replacement_query = ("select ''");
+        bool should_replace = false;
 
-    bool should_replace = false;
-    // Translate query from MySQL to ClickHouse.
-    // This is a temporary workaround until ClickHouse supports the syntax "@@var_name".
-    if (std::string(payload.position(), payload.buffer().end()) == "select @@version_comment limit 1")  // MariaDB client starts session with that query
-    {
-        should_replace = true;
-    }
+        // Translate query from MySQL to ClickHouse.
+        // This is a temporary workaround until ClickHouse supports the syntax "@@var_name".
+        if (query == "select @@version_comment limit 1")  // MariaDB client starts session with that query
+        {
+            should_replace = true;
+        }
+        // This is a workaround in order to support adding ClickHouse to MySQL using federated server.
+        if (0 == strncasecmp("SHOW TABLE STATUS LIKE", query.c_str(), 22))
+        {
+            should_replace = true;
+            replacement_query = boost::replace_all_copy(query, "SHOW TABLE STATUS LIKE ", "SELECT name AS Name, engine AS Engine, '10' AS Version, 'Dynamic' AS Row_format, 0 AS Rows, 0 AS Avg_row_length, 0 AS Data_length, 0 AS Max_data_length, 0 AS Index_length, 0 AS Data_free, 'NULL' AS Auto_increment, metadata_modification_time AS Create_time, metadata_modification_time AS Update_time, metadata_modification_time AS Check_time, 'utf8_bin' AS Collation, 'NULL' AS Checksum, '' AS Create_options, '' AS Comment FROM system.tables WHERE name=");
```

Comment:
I would split the third argument into several lines.

Reply:
fixed